shell/platform/linux/fl_standard_message_codec_test.cc

```diff
@@ -0,0 +1,989 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_standard_message_codec.h"
+#include "flutter/shell/platform/linux/testing/fl_test.h"
+#include "gtest/gtest.h"
+
+// FIXME(robert-ancell) Have two cases where endianess matters
+// FIXME(robert-ancell) Add tests for strings/lists/maps with more than 253
+// elements
+
+// Encode a message using a FlStandardMessageCodec. Return a hex string with the
+// encoded binary output.
+static gchar* encode_message(FlValue* value) {
+  g_autoptr(FlStandardMessageCodec) codec = fl_standard_message_codec_new();
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(GBytes) message =
+      fl_message_codec_encode_message(FL_MESSAGE_CODEC(codec), value, &error);
+  EXPECT_NE(message, nullptr);
+  EXPECT_EQ(error, nullptr);
+
+  return bytes_to_hex_string(message);
+}
+
+// Decode a message using a FlStandardMessageCodec. The binary data is given in
+// the form of a hex string.
+static FlValue* decode_message(const char* hex_string) {
+  g_autoptr(FlStandardMessageCodec) codec = fl_standard_message_codec_new();
+  g_autoptr(GBytes) data = hex_string_to_bytes(hex_string);
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(FlValue) value =
+      fl_message_codec_decode_message(FL_MESSAGE_CODEC(codec), data, &error);
+  EXPECT_EQ(error, nullptr);
+  EXPECT_NE(value, nullptr);
+  return fl_value_ref(value);
+}
+
+// Decode a message using a FlStandardMessageCodec. The binary data is given in
+// the form of a hex string. Expect the given error.
+static void decode_error_value(const char* hex_string,
+                               GQuark domain,
+                               gint code) {
+  g_autoptr(FlStandardMessageCodec) codec = fl_standard_message_codec_new();
+  g_autoptr(GBytes) data = hex_string_to_bytes(hex_string);
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(FlValue) value =
+      fl_message_codec_decode_message(FL_MESSAGE_CODEC(codec), data, &error);
+  EXPECT_TRUE(value == nullptr);
+  EXPECT_TRUE(g_error_matches(error, domain, code));
+}
+
+TEST(FlStandardMessageCodecTest, EncodeNULL) {
+  g_autofree gchar* hex_string = encode_message(nullptr);
+  EXPECT_STREQ(hex_string, "00");
+}
+
+TEST(FlStandardMessageCodecTest, EncodeNull) {
```

Comment:
The naming distinction is very subtle here; how about "EncodeNullptr" and "EncodeNullValue"?